File: frontend/src/lib/global-platform.ts
export type SynaptiverseRole =
  | 'patient'
  | 'caregiver'
  | 'clinician'
  | 'facility'
  | 'diagnostic_service'
  | 'pharmacy'
  | 'emergency_transport'
  | 'payer'
  | 'network_admin'
  | 'synaptiverse_admin';

export type TextDirection = 'ltr' | 'rtl';

export type RegionPolicyProfile = {
  regionId: string;
  countryCode: string;
  displayName: string;
  defaultLanguage: string;
  supportedLanguages: string[];
  textDirection: TextDirection;
  timeZones: string[];
  currencyCode: string;
  measurementSystem: 'metric' | 'imperial' | 'mixed';
  emergencyGuidanceSource: 'verified_regional_configuration';
  emergencyContactConfigured: boolean;
  sensitiveDataResidency: 'region_scoped' | 'country_scoped';
  crossBorderTransferRequiresApproval: boolean;
  clinicianJurisdictionRequired: boolean;
};

export type RoutingFactor =
  | 'clinical_capability'
  | 'urgency'
  | 'travel_time'
  | 'capacity'
  | 'operating_status'
  | 'specialist_availability'
  | 'payer_coverage'
  | 'jurisdiction';

export type ServiceBoundary = {
  service: string;
  frontendResponsibility: string;
  backendResponsibility: string;
};

export const globalClinicalBoundary = {
  positioning: 'Right care. Right facility. Right time.',
  patientFacingDisclaimer:
    'SynaptiVerse supports care coordination and does not diagnose, replace a clinician, or replace emergency services.',
  safetyRules: [
    'Never hardcode emergency numbers in product UI.',
    'Do not encode personal or health information in QR codes.',
    'Render triage output as explainable recommendations requiring clinical validation.',
    'Keep identifiable patient and clinical data scoped by tenant, region, and residency policy.',
    'Treat cross-border care access as an explicit consented and audited workflow.',
  ],
} as const;

export const demoRegionPolicyProfiles: RegionPolicyProfile[] = [
  {
    regionId: 'demo-global',
    countryCode: 'XX',
    displayName: 'Global Demo Region',
    defaultLanguage: 'en',
    supportedLanguages: ['en', 'qps-demo'],
    textDirection: 'ltr',
    timeZones: ['UTC'],
    currencyCode: 'USD',
    measurementSystem: 'metric',
    emergencyGuidanceSource: 'verified_regional_configuration',
    emergencyContactConfigured: false,
    sensitiveDataResidency: 'region_scoped',
    crossBorderTransferRequiresApproval: true,
    clinicianJurisdictionRequired: true,
  },
];

export const routingFactors: RoutingFactor[] = [
  'clinical_capability',
  'urgency',
  'travel_time',
  'capacity',
  'operating_status',
  'specialist_availability',
  'payer_coverage',
  'jurisdiction',
];

export const serviceBoundaries: ServiceBoundary[] = [
  {
    service: 'identity_and_auth',
    frontendResponsibility: 'Render role-aware sessions, redirects, and secure logout flows.',
    backendResponsibility: 'Own credentials, JWT/session refresh, MFA, tenant context, and role authorization.',
  },
  {
    service: 'triage_and_routing',
    frontendResponsibility: 'Collect guided intake, display explainable routing recommendations, and show loading/error states.',
    backendResponsibility: 'Own rules, jurisdiction policy, routing scores, appointment creation, and audit events.',
  },
  {
    service: 'regional_configuration',
    frontendResponsibility: 'Render locale-aware terminology, formatting, and emergency guidance from configuration.',
    backendResponsibility: 'Validate and store verified region policy profiles and regional data residency rules.',
  },
  {
    service: 'care_collaboration',
    frontendResponsibility: 'Show referrals, specialist exchange, records release, and notification states from APIs.',
    backendResponsibility: 'Own permissions, consent, cross-organization handoffs, records exchange, and websocket delivery.',
  },
  {
    service: 'payments_and_payers',
    frontendResponsibility: 'Display plan, authorization, and payment statuses returned by APIs.',
    backendResponsibility: 'Own payment adapters, payer verification, claims workflows, webhooks, and reconciliation.',
  },
];

export const roleWorkspaceRoutes: Record<SynaptiverseRole, string> = {
  patient: '/dashboard',
  caregiver: '/dashboard',
  clinician: '/specialist/dashboard',
  facility: '/hospital/dashboard',
  diagnostic_service: '/lab/dashboard',
  pharmacy: '/pharmacy/dashboard',
  emergency_transport: '/moh/dashboard',
  payer: '/hmo/dashboard',
  network_admin: '/moh/dashboard',
  synaptiverse_admin: '/dashboard/admin',
};

File: backend/app/global_platform.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


TextDirection = Literal["ltr", "rtl"]
MeasurementSystem = Literal["metric", "imperial", "mixed"]
DataResidencyScope = Literal["region_scoped", "country_scoped"]


class RegionPolicyProfile(BaseModel):
    region_id: str = Field(min_length=2, max_length=64)
    country_code: str = Field(min_length=2, max_length=3)
    display_name: str = Field(min_length=2, max_length=128)
    default_language: str = Field(min_length=2, max_length=16)
    supported_languages: list[str] = Field(default_factory=lambda: ["en"])
    text_direction: TextDirection = "ltr"
    time_zones: list[str] = Field(default_factory=lambda: ["UTC"])
    currency_code: str = Field(min_length=3, max_length=3)
    measurement_system: MeasurementSystem = "metric"
    emergency_guidance_source: Literal["verified_regional_configuration"] = "verified_regional_configuration"
    emergency_contact_configured: bool = False
    sensitive_data_residency: DataResidencyScope = "region_scoped"
    cross_border_transfer_requires_approval: bool = True
    clinician_jurisdiction_required: bool = True
    consent_policy_summary: str | None = Field(default=None, max_length=2000)
    retention_policy_summary: str | None = Field(default=None, max_length=2000)


class RoutingPolicy(BaseModel):
    policy_id: str = Field(min_length=2, max_length=64)
    region_id: str = Field(min_length=2, max_length=64)
    enabled_factors: list[
        Literal[
            "clinical_capability",
            "urgency",
            "travel_time",
            "capacity",
            "operating_status",
            "specialist_availability",
            "payer_coverage",
            "jurisdiction",
        ]
    ]
    requires_clinician_review: bool = True
    thresholds_are_clinician_approved: bool = False


class ClinicalSafetyBoundary(BaseModel):
    positioning: str = "Right care. Right facility. Right time."
    disclaimer: str = (
        "ClinicalFlow supports care coordination and does not diagnose, replace a clinician, "
        "or replace emergency services."
    )
    never_hardcode_emergency_contacts: bool = True
    never_encode_phi_in_qr_codes: bool = True
    cross_border_access_requires_policy_check: bool = True
    triage_requires_clinical_validation_label: bool = True


DEMO_REGION_POLICY_PROFILE = RegionPolicyProfile(
    region_id="demo-global",
    country_code="XX",
    display_name="Global Demo Region",
    default_language="en",
    supported_languages=["en", "qps-demo"],
    currency_code="USD",
)


GLOBAL_CLINICAL_SAFETY_BOUNDARY = ClinicalSafetyBoundary()
